pkg/kotsadmsnapshot/backup.go

```diff
@@ -191,254 +201,512 @@ func CreateApplicationBackup(ctx context.Context, a *apptypes.App, isScheduled b
+// instance backup.
+func getInstanceBackupMetadata(ctx context.Context, k8sClient kubernetes.Interface, ctrlClient ctrlclient.Client, veleroClient veleroclientv1.VeleroV1Interface, cluster *downstreamtypes.Downstream, isScheduled bool) (instanceBackupMetadata, error) {
+	now := time.Now().UTC()
+	metadata := instanceBackupMetadata{
+		backupName:       fmt.Sprintf("instance-%d", now.UnixNano()),
+		backupReqestedAt: now,
+		kotsadmNamespace: util.PodNamespace,
+		apps:             make(map[string]appInstanceBackupMetadata, 0),
+		isScheduled:      isScheduled,
+	}
+
+	if cluster.SnapshotTTL != "" {
+		snapshotTTL, err := time.ParseDuration(cluster.SnapshotTTL)
+		if err != nil {
+			return metadata, errors.Wrap(err, "failed to parse snapshot ttl")
+		}
+		metadata.snapshotTTL = snapshotTTL
 	}
 
+	kotsadmVeleroBackendStorageLocation, err := kotssnapshot.FindBackupStoreLocation(ctx, k8sClient, veleroClient, metadata.kotsadmNamespace)
+	if err != nil {
+		return metadata, errors.Wrap(err, "failed to find backupstoragelocations")
+	} else if kotsadmVeleroBackendStorageLocation == nil {
+		return metadata, errors.New("no backup store location found")
+	}
+	metadata.backupStorageLocationNamespace = kotsadmVeleroBackendStorageLocation.Namespace
+
 	apps, err := store.GetStore().ListInstalledApps()
 	if err != nil {
-		return nil, errors.Wrap(err, "failed to list installed apps")
+		return metadata, errors.Wrap(err, "failed to list installed apps")
 	}
 
-	for _, a := range apps {
-		downstreams, err := store.GetStore().ListDownstreamsForApp(a.ID)
+	for _, app := range apps {
+		downstreams, err := store.GetStore().ListDownstreamsForApp(app.ID)
 		if err != nil {
-			return nil, errors.Wrapf(err, "failed to list downstreams for app %s", a.Slug)
+			return metadata, errors.Wrapf(err, "failed to list downstreams for app %s", app.Slug)
 		}
 
 		if len(downstreams) == 0 {
-			logger.Error(errors.Wrapf(err, "no downstreams found for app %s", a.Slug))
+			logger.Errorf("No downstreams found for app %s", app.Slug)
 			continue
 		}
 
-		parentSequence, err := store.GetStore().GetCurrentParentSequence(a.ID, downstreams[0].ClusterID)
+		parentSequence, err := store.GetStore().GetCurrentParentSequence(app.ID, downstreams[0].ClusterID)
 		if err != nil {
-			return nil, errors.Wrapf(err, "failed to get current downstream parent sequence for app %s", a.Slug)
+			return metadata, errors.Wrapf(err, "failed to get current downstream parent sequence for app %s", app.Slug)
 		}
 		if parentSequence == -1 {
 			// no version is deployed for this app yet
 			continue
 		}
 
-		archiveDir, err := ioutil.TempDir("", "kotsadm")
+		archiveDir, err := os.MkdirTemp("", "kotsadm")
 		if err != nil {
-			return nil, errors.Wrapf(err, "failed to create temp dir for app %s", a.Slug)
+			return metadata, errors.Wrapf(err, "failed to create temp dir for app %s", app.Slug)
 		}
-		defer os.RemoveAll(archiveDir)
+		defer func() {
+			_ = os.RemoveAll(archiveDir)
```

Comment:
just curious, how is this different from just `defer os.RemoveAll(archiveDir)`?

Reply:
I really like the errcheck lint rule and I'd love to enable it

> Errcheck is a program for checking for unchecked errors in Go code. These unchecked errors can be critical bugs in some cases.

https://golangci-lint.run/usage/linters/#errcheck

Reply:
ah ok